x-pack/plugins/security_solution/public/common/components/navigation/solution_grouped_nav/solution_grouped_nav.tsx

```diff
@@ -0,0 +1,165 @@
+  items: NavItem[];
+  selectedId: string;
+  footerItems?: NavItem[];
+}
+type ActivePortalNav = string | null;
+
+export const SolutionGroupedNavComponent: React.FC<SolutionGroupedNavProps> = ({
+  items,
+  selectedId,
+  footerItems = [],
+}) => {
+  const { euiTheme } = useEuiTheme();
+  const styles = solutionGroupedNavStyles(euiTheme);
+  const isMobileSize = useIsWithinBreakpoints(['xs', 's']);
+
+  const [activePortalNavId, setActivePortalNavId] = useState<ActivePortalNav>(null);
+  const activePortalNavIdRef = useRef<ActivePortalNav>(null);
+
+  const openPortalNav = (navId: string) => {
+    activePortalNavIdRef.current = navId;
+    setActivePortalNavId(navId);
+  };
+
+  const closePortalNav = () => {
+    activePortalNavIdRef.current = null;
+    setActivePortalNavId(null);
+  };
+
+  const onClosePortalNav = useCallback(() => {
+    const currentPortalNavId = activePortalNavIdRef.current;
+    // This event is triggered on outside click
+    setTimeout(() => {
+      // closing the side nav at the end of event loop to make sure it
+      // closes even if the active "sub nav" open button has been clicked (toggle),
+      // but it does not close if any other "sub nav" open button has been clicked
+      if (activePortalNavIdRef.current === currentPortalNavId) {
+        closePortalNav();
+      }
+    });
+  }, []);
+
+  const navItemsById = useMemo(
+    () =>
+      [...items, ...footerItems].reduce<
+        Record<string, { title: string; subItems: PortalNavItem[] }>
+      >((acc, navItem) => {
+        if (isDefaultNavItem(navItem) && navItem.items && navItem.items.length > 0) {
+          acc[navItem.id] = {
+            title: navItem.label,
+            subItems: navItem.items,
+          };
+        }
+        return acc;
+      }, {}),
+    [items, footerItems]
+  );
+
+  const portalNav = useMemo(() => {
+    if (activePortalNavId == null || !navItemsById[activePortalNavId]) {
+      return null;
+    }
+    const { subItems, title } = navItemsById[activePortalNavId];
+    return <SolutionGroupedNavPanel onClose={onClosePortalNav} items={subItems} title={title} />;
+  }, [activePortalNavId, navItemsById, onClosePortalNav]);
+
+  const renderNavItem = useCallback(
+    (navItem: NavItem) => {
+      if (isCustomNavItem(navItem)) {
+        return <Fragment key={navItem.id}>{navItem.render()}</Fragment>;
+      }
+      const { id, href, label, onClick } = navItem;
+      const isActive = activePortalNavId === id;
+      const isCurrentNav = selectedId === id;
+
+      const itemClassNames = classNames('solutionGroupedNavItem', {
+        'solutionGroupedNavItem--isActive': isActive,
+        'solutionGroupedNavItem--isPrimary': isCurrentNav,
+      });
+      const buttonClassNames = classNames('solutionGroupedNavItemButton');
+
+      return (
+        // eslint-disable-next-line @elastic/eui/href-or-on-click
+        <EuiLink key={id} href={href} onClick={onClick} color={isCurrentNav ? 'primary' : 'text'}>
```

Comment:
I am not sure if it is a good solution. But most places are using `LinkAnchor` instead of `EuiLink` to be able to provide `ref` and `onClick` props to an anchor without a TS error.

Reply:
Yes, it would make sense, but I prefer to not use our security solution link components directly here, just in case we move this navigation implementation to a shared package. 
But I am using them indirectly, these `href` and `onClick` values I am passing in the NavItem are defined using the `useGetSecuritySolutionLinkProps` hook I extracted from the `withSecuritySolutionLink`.